api/v1alpha1/s3userclaim_webhook.go

```diff
@@ -128,35 +136,32 @@ func (suc *S3UserClaim) ValidateDelete() error {
 	return nil
 }
 
-func validateQuota(suc *S3UserClaim) field.ErrorList {
+func validateQuota(suc *S3UserClaim, allErrsPointer *field.ErrorList) {
```

Comment:
The convention is generally to avoid Hungarian notation. (including pointer in the name of the variable).

Also, it's a good practice to check for nil-ability of a pointer before using it.

---

Also 😄, for slices a convention is to pass them by value and assign the return value of the function to the existing slice.

This is because when passing a slice to a function, we pass a copy of the header (or pointer) of the slice not the whole slice. There's an edge case though. When the underlying array should be resized, the slice will point to a new array. This is why we should return the slice and reassign the existing one.

That is, something like this:

```go
func validate(errList field.ErrorList) (field.ErrorList, error) {
	errList = append(errList, field.Forbidden(field.NewPath("sample/path"), "sample detail"))
	return errList, nil
}

func main() {
        errList := field.ErrorList{}
	errList, err = validate(errList)
	if err != nil {
		panic("Bad")
	}
}
```

Reply:
Many thanks for your rich explanation. I learned a lot!
Sure I'll do it.

Reply:
What about structs and maps? Shall we behave like slices to them or it's better to use pointers?

Reply:
Maps are the same.

Pointers are preferred for structs.

Reply:
Done